api/nodes/permissions.py

```diff
@@ -148,6 +180,29 @@ def has_object_permission(self, request, view, obj):
             return node.has_permission(auth.user, osf_permissions.ADMIN)
 
 
+class NodeGroupDetailPermissions(permissions.BasePermission):
+    """Permissions for node group detail - involving who can update the relationship
+    between a node and an OSF Group."""
+
+    acceptable_models = (OSFGroup, AbstractNode,)
+
+    def load_resource(self, context, view):
+        return AbstractNode.load(context[view.node_lookup_url_kwarg])
+
+    def has_object_permission(self, request, view, obj):
+        assert_resource_type(obj, self.acceptable_models)
+        auth = get_user_auth(request)
+        node = self.load_resource(request.parser_context['kwargs'], view)
+        if request.method in permissions.SAFE_METHODS:
+            return node.is_public or node.can_view(auth)
+        elif request.method == 'DELETE':
```

Comment:
Yes, we're removing the group from the node, not literally deleting the group.  I'm broadly thinking of this as a delete operation, we're removing permissions to the Node from the OSF Group's Django member group.  Plus, because DELETE and PATCH operations need different permissions, separating this way makes it easy to do this from a high-level. Finally, this is consistent with many of our other similar operations, deleting a contributor from a node for example. We're not technically deleting the user object, but we're removing the relationship between that user and the node, plus the user's permission to that node, and whether the user is bibliographic.